contracts/PodCastNFT.sol

```diff
@@ -227,14 +224,16 @@ contract PodCastNFT is ERC721, Ownable {
         return false;
     }
 
-    // TODO implement updateNFT func which holder can change their name, imageURL by them self
-    function updateOwnNFT(string memory _imageURI, string memory name) public {}
+    function updateOwnNFT(uint256 tokenId, string memory finalTokenUri) public {
+        require(ownerOf(tokenId) == msg.sender, "NOT NFT OWNER");
```

Comment:
uh, but updateNFT by only owner's arg is tokenId so maybe ok.

Reply:
https://github.com/henkaku-center/podcast-nft/pull/66/commits/c8ada75896ca0289ff1ac96fe890d4d16a198551

The first version implemented both. However, due to size limits, it became necessary to remove some of them.

OnlyHolder alone would be a problem; Holder-to-Holder would allow you to update other people's token URI.